File: Practica9/Java/Ejercicio 2/Cuadrado.java
import java.awt.*;
class Cuadrado extends Figura implements Coloreado {
    double lado;

    Cuadrado(double l, String c) {
        super(c);
        lado = l;
    }

    double area() {
        return lado * lado;
    }

    double perimetro() {
        return 4 * lado;
    }

    public String comoColorear() {
        return "Colorear los cuatro lados";
    }

    void dibujar(Graphics g, int x, int y) {
        g.setColor(Color.decode(color));
        g.fillRect(x, y, (int)lado, (int)lado);
    }
}

File: Practica9/Python/Ejercicio 2/Cuadrado.py
from Figura import Figura
from Coloreado import Coloreado
class Cuadrado(Figura, Coloreado):
    def __init__(self, lado, color):
        super().__init__(color)
        self.lado = lado

    def area(self):
        return self.lado * self.lado

    def perimetro(self):
        return 4 * self.lado

    def como_colorear(self):
        return "Colorear los cuatro lados"

    def __str__(self):
        return "Cuadrado - " + super().__str__() + \
               ", Área: " + repr(self.area()) + \
               ", Perímetro: " + repr(self.perimetro()) + \
               ", " + self.como_colorear()
